Makefile

```diff
@@ -169,6 +174,7 @@ clean: $(CLEAN_TARGETS) docpreview_clean
 	@rm -rf $(DEPSDIR)
 	@rm -rf lib/
 	@rm -rf bin/coverage.*.xml
+	@rm -rf $(CMDDIR) compile_commands.json
```

Comment:
Does it make sense to introduce `clean all`? `make clean` gets called relatively frequently. Cleaning up compile command file, which IDE depends on might not be nice.

Reply:
we could have our own `make mrproper` that removes compile_commands.json and depends on `clean` ?

Reply:
I am not aware of `mrproper` until now. Seems like linux kernel uses it. If its more standard/correct way of doing it, sure go for it.